src/components/login/Login.js

```diff
@@ -2,77 +2,215 @@ import React from "react";
 import apiUser from "../../services/apiUser";
 import "./Login.css";
 import { withRouter } from "react-router-dom";
+const crypto = require('crypto');
 
+function makeHash(Password) {
+  return crypto.createHash('sha256').update(Password, 'utf8').digest().toString('hex')
+}
 
-class Login extends React.Component{
-  constructor(props){
+class Login extends React.Component {
+  constructor(props) {
     super(props);
     this.state = {
       username: "",
-      password: ""
+      password: "",
+      email: "",
+      code: "",
+      token: "",
+      newPassword: "",
+      newPasswordConfirm: "",
+      stateSwitch: 0
     };
 
     this.handleChange = this.handleChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleEmail = this.handleEmail.bind(this);
+    this.handleCode = this.handleCode.bind(this);
+    this.handleChangePassword = this.handleChangePassword.bind(this);
   }
 
-    // BUSCAR TOKEN
-    // http://172.25.0.3:3000/sessions POST
-    // body.json = { "username": "admin",	"password": "123456" }
-    // retorna user e o token
+  // BUSCAR TOKEN
+  // http://172.25.0.3:3000/sessions POST
+  // body.json = { "username": "admin",	"password": "123456" }
+  // retorna user e o token
 
-    // BUSCAR USUARIO
-    // http://172.25.0.3:3000/users/id POST
-    // body.json = { "username": "admin" }
-    // retorna user
+  // BUSCAR USUARIO
+  // http://172.25.0.3:3000/users/id POST
+  // body.json = { "username": "admin" }
+  // retorna user
 
-    // CRIAR USUARIO
-    // http://172.25.0.3:3000/users POST
-    // body.json = { "username": "TESTE", 	"user_id": "4",	"name": "teste", "surname": "teste", "password": "123456" }
-    // retorna user 
+  // CRIAR USUARIO
+  // http://172.25.0.3:3000/users POST
+  // body.json = { "username": "TESTE", 	"user_id": "4",	"name": "teste", "surname": "teste", "password": "123456" }
+  // retorna user 
 
-  handleChange(event){
-    if(event.target.id === "username"){
-      this.setState({username: event.target.value});
+  handleChange(event) {
```

Comment:
Codacy found an issue: [Method 'handleChange' has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/fga-eps-mds/2020.1-VC_Gestor-FrontEnd/pullRequest?prid=6397155)